Development/nmos/node_server.cpp

```diff
@@ -113,6 +117,16 @@ namespace nmos
                 node_server.thread_functions.push_back([&, load_ca_certificates, system_changed] { nmos::node_system_behaviour_thread(node_model, load_ca_certificates, system_changed, gate); });
             }
 
+// only implement communication with OCSP server if http_listener supports OCSP stapling
+// cf. preprocessor conditions in nmos::make_http_listener_config
+#if !defined(_WIN32) || defined(CPPREST_FORCE_HTTP_LISTENER_ASIO)
+            if (server_secure)
```

Comment:
Is there a way of disabling the OCSP implementation? Would that be a good idea?

Reply:
I thought this was really driven by the client, if the client is asking for it in the handshake we should always try to serve it.

We could add a new `ocsp_enable` in the settings to indicate whether to insert/or not of the `ocsp_behaviour_thread` and default the value to true.

Reply:
Yes, we can move the `ocsp_behaviour_thread` to the node implementation and when an OCSP response is received at the `ocsp_behaviour_thread`, then it executes the on_receive_ocsp_response callback on nmos to cache the OCSP response.

Callback as

```
namespace nmos
{
    namespace experimental
    {
        struct ocsp_state;
    }

    // callback to supply OCSP response in DER format
    // this callback is executed when receiving the OCSP response
    // this callback should not throw exceptions
    typedef std::function<void(const std::vector<uint8_t>&)> ocsp_response_handler;

    // construct callback to receive OCSP response
    ocsp_response_handler make_ocsp_response_handler(nmos::experimental::ocsp_state& ocsp_state, slog::base_gate& gate);
}
```

Reply:
I was thinking sorta like this...

```C++
namespace nmos
{
    namespace experimental
    {
        struct ocsp_state;
    }

    // callback to return OCSP response in DER format
    // this callback is executed for each new TLS connection
    // this callback should not throw exceptions
    typedef std::function<std::vector<uint8_t>()> ocsp_response_handler;

    // construct callback to return OCSP response
    ocsp_response_handler make_ocsp_response_handler(nmos::experimental::ocsp_state& ocsp_state, slog::base_gate& gate);
}
```